src/libraries/System.Text.Json/src/System/Text/Json/Writer/JsonWriterOptions.cs

```diff
@@ -93,9 +135,11 @@ public bool SkipValidation
             }
         }
 
-        internal bool IndentedOrNotSkipValidation => _optionsMask != SkipValidationBit; // Equivalent to: Indented || !SkipValidation;
+        internal bool IndentedOrNotSkipValidation => Indented || !SkipValidation;
 
         private const int IndentBit = 1;
         private const int SkipValidationBit = 2;
+        private const int IndentCharacterBit = 4;
+        private const int IndentSizeMask = JsonConstants.MaximumIndentSize << 3;
```

Comment:
Could this break if somebody changed the `MaximumIndentSize` value? Perhaps add a debug assertion validating that it does not?

Reply:
It would not break until MaximumIndentSize << 3 is <= int.MaxValue or if, in between, someone encodes a new value in the free bits.
Nevertheless, I have tried to add a debug assertion, but I get the following error:
```
error CS8520: The given expression always matches the provided constant.
```
Is there an alternative way in such a specific case?

Reply:
If everything else fails, a comment in the definition of the original constant asking users to also pay attention to this location might work.

Reply:
Comment added.